packages/core/src/rest/rest-request-builder.ts

```diff
@@ -0,0 +1,21 @@
+import { AxiosRequestConfig } from 'axios';
+import { Destination, DestinationNameAndJwt } from '../scp-cf';
+import { buildAxiosRequestConfig } from '../http-client';
+import { RestRequestConfig } from './rest-request-config';
+
+export abstract class RestRequestBuilder {
+  constructor(public requestConfig: RestRequestConfig) {}
+
+  withCustomHeaders(headers: Record<string, string>): this {
+    this.requestConfig.addCustomHeaders(headers);
+    return this;
+  }
+
+  buildRequestConfig(
+    destination: Destination | DestinationNameAndJwt
+  ): Promise<AxiosRequestConfig> {
+    return buildAxiosRequestConfig(destination, {
```

Comment:
I think the CSRF header are not fetched automatically. You could hand them in via the custom headers of course but it would be nice to include the header automatically as in the `headers()` method of the OData request.

Reply:
https://github.wdf.sap.corp/MA/scp-rocket/issues/1525